src/v4/mod.rs

```diff
@@ -1,4 +1,83 @@
-//! DHCPv4 Message type
+//! # DHCPv4
+//!
+//! This module provides types and utility functions for encoding/decoding a DHCPv4 message.
+//!
+//! ## Example - constructing messages
+//!
+//! ```rust
+//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! use dhcproto::{v4, Encodable, Encoder};
```

Comment:
This feels odd,  `use` inside the `{ }` of `main()`.

Consider to have lines swap, to have this
```
//! use dhcproto::{v4, Encodable, Encoder};
//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Reply:
Thanks for reviewing! The `#` prevents that line from actually showing in the docs, so I'm not too worried about it.

While I have you, any thoughts on the actual examples?